ibis/expr/operations/temporal.py

```diff
@@ -213,6 +213,45 @@ class Date(UnaryOp):
     output_type = rlz.shape_like('arg', dt.date)
 
 
+@public
+class DateFromYMD(ValueOp):
+    year = rlz.integer
+    month = rlz.integer
+    day = rlz.integer
+    output_type = rlz.shape_like('args', dt.date)
+
+
+@public
+class TimeFromHMS(ValueOp):
+    hours = rlz.integer
+    minutes = rlz.integer
+    seconds = rlz.integer
+    output_type = rlz.shape_like('args', dt.time)
+
+
+@public
+class TimestampFromYMDHMS(ValueOp):
+    year = rlz.integer
+    month = rlz.integer
+    day = rlz.integer
+    hours = rlz.integer
+    minutes = rlz.integer
+    seconds = rlz.integer
+    timezone = rlz.optional(rlz.string)
+    output_type = rlz.shape_like('args', dt.timestamp)
+
+    @property
+    def arg(self):
```

Comment:
I don't think this is necessary. You should be able to access everything as `.args[:-1]`, or preferably as each individual component so the usage of the operations arguments isn't sensitive to argument declaration order.

Reply:
`varargs(...)` requires an `.arg`.  Changing those to `fixed_arity(..., 6)` doesn't work because of the extra timezone field (unused currently but seemed important to include in the API, and then to capture in the op).  I got everything to work this way with the `arg` property; is this actually incorrect?

Reply:
How do you feel about changing `fixed_arity` to trim the number of args instead of raising an error?  Or making a similar helper?  I could do it inline but that seems too painful.

Reply:
I'd like to avoid giving `fixed_arity` any convenience functionality. It's only there to make the simplest cases convenient, and everything else has to be explicit.

Reply:
You can also write a translation rule and remove the `.arg` property and also avoid having to change `fixed_arity` or add a similar helper:

```python
def rule(t, expr):
	return sa.func.make_timestamp(*map(t.translate, expr.op().args[:-1]))
```